### DISCUSSION TITLE
New Service Requests / Suggestions

### DISCUSSION THREAD

**Discussion by kn0wmad** [Start9Labs/start-os#386] [Ideas]
Just as the title suggests.  This acts in the same way as the unofficial Embassy:Voice (https://s9.altweb.me/), and is just another platform for folks to voice their ideas.  The two should compliment each other in my opinion

I'll start the discussion with a vote for Nextcloud, currently the most requested on Embassy:Voice

**Comment by k0gen** [comment]
LndHUB for BlueWallet
https://bluewallet.io/lndhub/

A Lightning Hub - Ideal for hosted shared services or groups, families and friends that want to share their own node.

**Reply by kn0wmad** [reply]
Now available in LNBits.

**Comment by midizzz** [comment]
Hi. Love start9!
I am about to try to run zfs on the root drive. Maybe you can tell me if this has worked well in the past? Would be awesome to have a guide.

**Reply by kn0wmad** [reply]
We do not support custom filesystems at this time.  You are welcome to hack around, but you will likely have unintended consequences.

**Reply by midizzz** [reply]
Thanks good to know. I will hold off. Considering it's coming as standard on ubuntu now it might trickle through to you I hope. Again this is a routing node type wish.

**Comment by midizzz** [comment]
Hi. Cyberpunk fanatic fan again.
Imagine if you had a VLS function on start9 an app or a feature that could be turned on to connect to the signing device. Would really start to make the node way more pro grade than the umbrel.

**Reply by kn0wmad** [reply]
We are already more "pro grade."  Umbrel is just a docker-compose scheme with a UI.  This will be glaringly obvious following the v040 release of StartOS, when we will be directly competing with Big Tech cloud services.  In regard to VLS, you generally do not want to run signers on your server for security reasons. It would be better to use VLS in a supported wallet, which you then connect back to your Bitcoin/Lightning nodes on StartOS.

**Reply by midizzz** [reply]
Who said umbrel? Anyway i agree. Their pihole is not working atm also. Grrr.. with vls i dod not mean on the server, i mean separated but connected to. the idea would allow for start 9 to be used for larger nodes, and in that case the other apps would not be used.

**Reply by kn0wmad** [reply]
> Hi. Cyberpunk fanatic fan again. Imagine if you had a VLS function on start9 an app or a feature that could be turned on to connect to the signing device. Would really start to make the node way more pro grade than the umbrel.

^^ You did, haha :)

For services outside of what we officially support, the best place to begin a request is with the upstream devs.  Have you spoken to the VLS team at all?

**Reply by Cyberpunk-Hardware** [reply]
ahaha.. sorry, I was focused on the technical aspect for VLS. I also know Sphinx were working on this. I wonder if it would work through all the channels on the node... I haven't spoken to them. I will keep you up to date is something starts happening and if you could do the same.

**Comment by MEdGalo** [comment]
Hi,

1. Please consider adding Fulcrum as a replacement and/or additional choice for Electrs I posted this a while ago here as well (https://github.com/Start9Labs/service-pipeline/issues/71). Benchmarks show that it's much more performant than Electrs.

2. Since the webUI place an important role in startOS please consider adding 2FA to the login screen for the webUI.

**Reply by kn0wmad** [reply]
1. This was packaged over a year ago.  We've been waiting on Fulcrum to [fix a bug](https://github.com/cculianu/Fulcrum/issues/41) that they've known about for several years.  They claim it is coming in 2.0.

2. This is a bit of a low priority atm due to the lack of remote discoverability of your webUI unless you intentionally share it.  This will likely become more urgent after the 0.3.6 release.  You can [track the feature request here](https://github.com/Start9Labs/start-os/issues/1458).

**Reply by MEdGalo** [reply]
1. I've never had this bug while running Fulcrum on my raspiblitz previously for almost two years. It's seems to (almost) only happen after an ungraceful shutdown while during syncing the index. And the consequence would only be that you would have to reindex Fulcrum. While after an ungraceful shutdown you could also corrupt your LND/CLN database and have much more trouble as a result. Should users then also not be protected from using LND/CLN?
Wouldn't it be possible to release it under the community marketplace instead of the official start9 marketplace (and maybe add some extra warning to the page)?

2. The discoverability should be low indeed, but I remember this case with Umbrel a while ago that users leaked their .onion address through the Brave browser somehow. It's nice to have this extra layer of security to protect the user if they somehow 'leak' their webUI address.

**Reply by kn0wmad** [reply]
This was not our experience.  The bug was unacceptable during our testing, and that of others.  This is not at all the experience with CLN/LND, and there is a high demand for both (unlike fulcrum).  We have had never experienced this sort of issue with either node implementation.  [Here is the fulcrum wrapper repo](https://github.com/chrisguida/fulcrum-wrapper), you are more than welcome to play with it, or even publish it to your own website or registry.  We can not (and don't want to) prevent anyone from running whatever they would like, and are only picky about the software that we publish, as it also comes with our support.

**Reply by leCoq42** [reply]
Has the bug that you're describing been fixed in in fulcrum 1.10?

"Bugfix: In rare circumstances, if the "Download Blocks Task" is interrupted / gets an error, one of the 3 BitcoinD client threads may go out to lunch and never become usable again to do work. Symptoms would include it spamming "Reconnecting ..." messages to the log every 5 seconds. This has been corrected."

 https://github.com/cculianu/Fulcrum/releases/tag/v1.10.0

**Reply by leCoq42** [reply]
It should be fixed in version 2.0:
https://github.com/cculianu/Fulcrum/releases/tag/v2.0.0

**Comment by Reformer3322** [comment]
Hello,

I would like to suggest a new supported feature. The ability to run 2 instances of the same software would be really helpful in certain situations. Ex. running multiple instances of DATUM, LND.

I do not have technical capabilities to do this myself and couldn't find a supported way to do this. 

Thanks in advance to any response!
